### ISSUE TITLE
empty blank div on responsive slick when using slickAdd() on empty carousel

### ISSUE THREAD

**Issue by elubin** [kenwheeler/slick#931]
Great tool!

when i configure slick with responsive control, start with an empty carousel and fill it with slickAdd (after I get a response from my server), I have 2 issues.  First I get an extra empty div in the carousel which pushes off my first entry.  Second, not clear in the fiddle, the carousel div grows twice as tall and completely destroys the layout.

http://jsfiddle.net/bof71xxn/

**Comment by kenwheeler** [comment]
I'm not seeing slickAdd getting used here?

**Comment by elubin** [comment]
sorry, wrong fiddle.  http://jsfiddle.net/zp0yzw0f/

when the result screen is large enough, it's fine.  once it's below the responsive, if you run the page again, you'll see a blank empty div.

for now, i was able to fix it in my code by removing the element before i add any,...

**Comment by kenwheeler** [comment]
I think this is happening because slick can't initialize fast enough when you are using slickAdd in a loop

**Comment by elubin** [comment]
not sure if you mean initialization of the control or each element as they are added?

in real life, i am actually making a call to a remote api before populating the results into slick.  the control itself would have plenty of time to initialize.  however, the results are still added in a loop.

(and THANKS for your quick responses)

**Comment by elubin** [comment]
...and why ONLY when responsive is turned on?

**Comment by kenwheeler** [comment]
Ok, what's happening is, if you look at the markup, slick is trying to slick the interior slides. If you add specificity and declare your slide using the slide option, it fixes it. Add 'slide: ".photo"' to your option sets.

**Comment by elubin** [comment]
excellent!  that solves it!  thank you!  (maybe add something like that to your docs under responsive)

**Comment by samwx** [comment]
THANKSS!!!!111eleven!!1

**Comment by MikeInSwitzerland** [comment]
What I was finding was that if I created a slick control, but later added a new <img> to my slick list of images, it seemed to just show the first row of the new image, but no more.  If I resized my browser window, then the rest of the photo would magically appear.

My solution was this:  After adding any images to a scrolling slick, I would measure the height of my slick control, and just check that all images within that slick had that height.

                    var height = $('.cssCarousel').height();
                    $('.slick-track img').css("height", height + "px");

Odd, but it worked for me.
